pkg/asset/cluster/metadata.go

```diff
@@ -0,0 +1,107 @@
+package cluster
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/cluster/aws"
+	"github.com/openshift/installer/pkg/asset/cluster/libvirt"
+	"github.com/openshift/installer/pkg/asset/cluster/openstack"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	"github.com/pkg/errors"
+)
+
+const (
+	metadataFileName = "metadata.json"
+)
+
+// Metadata contains information needed to destroy clusters.
+type Metadata struct {
+	file *asset.File
+}
+
+var _ asset.WritableAsset = (*Metadata)(nil)
+
+// Name returns the human-friendly name of the asset.
+func (m *Metadata) Name() string {
+	return "Metadata"
+}
+
+// Dependencies returns the direct dependencies for the metadata
+// asset.
+func (m *Metadata) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.ClusterID{},
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the metadata asset.
+func (m *Metadata) Generate(parents asset.Parents) (err error) {
+	clusterID := &installconfig.ClusterID{}
+	installConfig := &installconfig.InstallConfig{}
+	parents.Get(clusterID, installConfig)
+
+	if installConfig.Config.Platform.None != nil {
+		return errors.New("metadata cannot be created with platform set to 'none'")
+	}
+
+	metadata := &types.ClusterMetadata{
+		ClusterName: installConfig.Config.ObjectMeta.Name,
+		ClusterID:   clusterID.ClusterID,
+	}
+
+	switch {
+	case installConfig.Config.Platform.AWS != nil:
+		metadata.ClusterPlatformMetadata.AWS = aws.Metadata(clusterID.ClusterID, installConfig.Config)
+	case installConfig.Config.Platform.Libvirt != nil:
+		metadata.ClusterPlatformMetadata.Libvirt = libvirt.Metadata(installConfig.Config)
+	case installConfig.Config.Platform.OpenStack != nil:
+		metadata.ClusterPlatformMetadata.OpenStack = openstack.Metadata(clusterID.ClusterID, installConfig.Config)
+	default:
```

Comment:
A case needs to be added for `None` platform, to allow success of the `ignition-configs` target.

Reply:
> A case needs to be added for `None` platform...

With e523949, we exit before getting this far, which should address your concern here without actually adding another case.